examples/userdemo/colormap_interactive_adjustment.py

```diff
@@ -0,0 +1,52 @@
+"""
+========================================
+Interactive Adjustment of Colormap Range
+========================================
+
+Demonstration of using colorbar and picker functionality to make an
+interactively adjustable colorbar widget.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.backend_bases import MouseButton
+
+
+def pick_fn(event):
+    adjust_colorbar(event.mouseevent)
+
+
+def motion_fn(mouseevent):
+    if mouseevent.inaxes is colorbar.ax.axes:
+        adjust_colorbar(mouseevent)
+
+
+def adjust_colorbar(mouseevent):
+    if mouseevent.button == MouseButton.LEFT:
+        colorbar.norm.vmax = max(mouseevent.ydata, colorbar.norm.vmin)
+    elif mouseevent.button == MouseButton.RIGHT:
+        colorbar.norm.vmin = min(mouseevent.ydata, colorbar.norm.vmax)
+    else:
+        # discard all others
+        return
+
+    canvas.draw_idle()
+
+
+fig, ax = plt.subplots()
+canvas = fig.canvas
+arr = np.random.random((100, 100))
+axesimage = plt.imshow(arr)
```

Comment:
AFAICS, the data range mapping to the full color scale is reduced by the interactive selection. As a result, we'll have over/under data. Unfortunately the default colormap marks these with the regular max/min colorbar values. This is quite confusing in the scaling context. I propose to mark out-of scale values:

```suggestion
cmap = plt.get_cmap('viridis').with_extremes(under='darkgrey', over='grey')
axesimage = plt.imshow(arr, cmap=cmap)
```